packages/app-admin-ui/client/pages/Item/index.js

```diff
@@ -388,96 +388,96 @@ const ItemNotFound = ({ adminPath, errorMessage, list }) => (
 const ItemPage = ({ list, itemId, adminPath, getListByKey }) => {
   const itemQuery = list.getItemQuery(itemId);
   const { addToast } = useToasts();
+  const { loading, error, data, refetch } = useQuery(itemQuery, {
+    // network-only because the data we mutate with is important for display in the UI, and may be different than what's in the cache
+    fetchPolicy: 'network-only',
+    errorPolicy: 'all',
+  });
+  const [updateItem, { loading: updateInProgress, error: updateError }] = useMutation(
+    list.updateMutation,
+    {
+      onError: updateError => {
+        const [title, ...rest] = updateError.message.split(/\:/);
+        const toastContent = rest.length ? (
+          <div>
+            <strong>{title.trim()}</strong>
+            <div>{rest.join('').trim()}</div>
+          </div>
+        ) : (
+          updateError.message
+        );
+
+        addToast(toastContent, {
+          appearance: 'error',
+        });
+      },
+    }
+  );
+
   return (
     <Suspense fallback={<PageLoading />}>
-      {/* network-only because the data we mutate with is important for display
-          in the UI, and may be different than what's in the cache */}
-      <Query query={itemQuery} fetchPolicy="network-only" errorPolicy="all">
-        {({ loading, error, data, refetch }) => {
-          // Now that the network request for data has been triggered, we
-          // try to initialise the fields. They are Suspense capable, so may
-          // throw Promises which will be caught by the above <Suspense>
-          captureSuspensePromises(
-            list.fields
-              .filter(({ isPrimaryKey }) => !isPrimaryKey)
-              .filter(({ maybeAccess }) => !!maybeAccess.update)
-              .map(field => () => field.initFieldView())
+      {(() => {
```

Comment:
This is needed as we can not run [these 2 linee](https://github.com/keystonejs/keystone-5/pull/1701/files#diff-18fd6ba6fe6ea61ebfb3250f37792127R449) outside of `<Suspense />`, it throws promise which would be either caught and waited inside hook or let suspense take care of that.

I am sure there can be improvement which I am unable to think of right now.

Reply:
@jesstelford @timleslie please suggest improvement to this block of code.